services/task-svc/internal/bed/bed.go

```diff
@@ -211,6 +214,12 @@ func (ServiceServer) UpdateBed(ctx context.Context, req *pb.UpdateBedRequest) (*
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
+	_ = events.DispatchBedUpdatedEvent(ctx, bedID, req.GetName())
+
+	if roomId.Valid {
+		_ = events.DispatchBedMovedToAnotherRoomEvent(ctx, bedID, roomId.UUID.String())
+	}
```

Comment:
Does it make sense to always dispatch the `BedMovedToAnotherRoomEvent` if the `roomId` is given & valid?

This means the `BedMovedToAnotherRoomEvent` will be dispatched even if it hasn't really changed but is inside the request, right?

Reply:
On the technical side: same goes for the `.UpdateBed`. We never check for diffs but just applying the requested changes by the user.
On the business logic side: There are two sides to this i guess:
- We can dispatch the event only if the value really changes
- We can dispatch the event every time the user dispatched the change
In event sourcing, both cases should result in the same state of our aggregate if our event handlers are idempotent. If the event gets consumed by services that are outside of this bounded context (task-svc) via an integration event (svc-to-svc events) the receiver can be sure that the bed actually moved.

I guess its a design decision? We should discuss this in the actual event sourcing implementation at https://github.com/helpwave/services/pull/587. I would close this conversation :)